uPortal-portlets/src/main/java/org/apereo/portal/portlets/marketplace/PortletMarketplaceController.java

```diff
@@ -379,8 +379,21 @@ private void setUpInitialView(
             categoryList.remove(this.portletCategoryRegistry.getTopLevelPortletCategory());
         }
 
+        logger.debug("initialFilter: {}", initialFilter);
+        String filter =
+                initialFilter == null
+                        ? null
+                        : categoryList
+                                .stream()
+                                .parallel()
```

Comment:
:notebook: https://dzone.com/articles/think-twice-using-java-8

Reply:
The work is minimal against a final variable, so no concerns of runaway threads. Good point, though.